lib/dockerregistry/inventory.go

```diff
@@ -106,12 +106,23 @@ func MakeSyncContext(
 	return sc, nil
 }
 
+func isPathRelative(rootPath string, relativePath string) error {
+	_, err := filepath.Rel(rootPath, relativePath)
```

Comment:
The docs for `Rel` https://golang.org/pkg/path/filepath/#Rel have an example, and in one case `Rel("/a", "/b/c")` returns `../b/c` (no error, even though the two paths are outside of each other).